### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by 3ipp0**
New PR from dev to main.

### CODE REVIEW

File: Software/CT.BusinessLogic/Services/SerialSettings.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace CT.BusinessLogic.Services
+{
+    class SerialSettings
```
Comment (strange-v) [inline Software/CT.BusinessLogic/Services/SerialSettings.cs:3]: It isn't a service, why it placed in the services directory?

File: Software/CT.Web/Pages/Index.razor.cs [hunk 2]
```diff
@@ -1,14 +1,68 @@
...
+        public string totalKm { get; set; } = "";
+        public string message { get; set; } = "";
+        public string canCommand { get; set; } = "";
 
+        protected override async Task OnInitializedAsync()
+        {
+            Notifier.Notify += GetValue;
+            try
+            {
+                await Task.Delay(300);
```
Comment (strange-v) [inline Software/CT.Web/Pages/Index.razor.cs:35]: o_O

File: Software/CT.BusinessLogic/Services/CanHandlers/HandlersSequence.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
+﻿using CT.BusinessLogic.Entities;
+using CT.BusinessLogic.Interfaces;
+
+namespace CT.BusinessLogic.Services.CanHandlers
+{
+    public class HandlersSequence : AbstractHandler
```
Comment (strange-v) [inline Software/CT.BusinessLogic/Services/CanHandlers/HandlersSequence.cs:6]: I'd create a separate service from this with bigger responsibilities. Let's discuss this at the meeting.

**Review by strange-v — CHANGES_REQUESTED** [review]

File: Software/CT.BusinessLogic/Services/SerialService.cs [hunk 4]
```diff
@@ -1,80 +1,72 @@
...
+                    if (Notify != null)
+                    {
+                        Notify.Invoke(value.ToString());         
+                    }
+                }
+            }
+            catch
             {
-                string msg = ex.Message;
+                //ToDo: Handle Exception VF
```
[the remark this answers is not in the document]
Comment (3ipp0) [reply Software/CT.BusinessLogic/Services/SerialService.cs:68]: :)

File: Software/CT.Web/Pages/Index.razor [hunk 5]
```diff
@@ -1,43 +1,120 @@
...
-        <div class="digital-monitor-center net-center">@ScaleValue</div>
+    
+<div class="container-fluid">
+    <div class="row">
+        <div class="col-md-10">
+            <h1 class="display-4">Car Telemetry</h1>
+            <h2 style="color:darkred;">@message</h2>
+        </div>
+        <div class="col-md-2">
+            <h1>@DateTime.Now.ToString("HH:mm")</h1>
```
[the remark this answers is not in the document]
Comment (Anbashman) [reply Software/CT.Web/Pages/Index.razor:12]: I don't know how this magic works, but it updates itself

File: Software/CT.BusinessLogic/Services/SerialSettings.cs (hunk 1 above)
Comment (Anbashman) [reply Software/CT.BusinessLogic/Services/SerialSettings.cs:3]: This file is not required

File: Hardware/CAN-receiver/CAN-receiver.ino [hunk 6]
```diff
@@ -16,28 +16,28 @@ eXoCAN can;
...
 
 uint32_t last = 0;
 char output_str[28];
-char output_format[]="%.3X %.2X %.2X %.2X %.2X %.2X %.2X %.2X %.2X";
+char output_format[] = "%.3X %.2X %.2X %.2X %.2X %.2X %.2X %.2X %.2X";
 
 void loop() {
   // poll for rx
-  if (can.receive(id, fltIdx, rxbytes) > -1) { 
+  if (can.receive(id, fltIdx, rxbytes) > -1) {
```
[the remark this answers is not in the document]
Comment (f1andrew) [reply Hardware/CAN-receiver/CAN-receiver.ino:30]: -1 is default return value of receive() function in exoCAN library. This approach taken from library example.

**Pull request merged**